pyro/infer/mcmc/logger.py

```diff
@@ -0,0 +1,111 @@
+import json
+import logging
+import sys
+from collections import OrderedDict
+
+from tqdm.autonotebook import tqdm
+
+LOG_MSG = "LOG"
+TQDM_MSG = "TQDM"
+
+
+class TqdmHandler(logging.StreamHandler):
+    """
+    Handler that synchronizes the log output with the
+    :class:`~tqdm.tqdm` progress bar.
```

Comment:
We need to redirect all logging to `tqdm.write`, without which progress bar and logging output will be interspersed.